repose-aggregator/components/services/datastore-service/datastore-service-api/src/main/scala/org/openrepose/core/services/datastore/types/PatchableSet.scala

```diff
@@ -30,22 +30,17 @@ import scala.collection.mutable
 class PatchableSet[A] extends mutable.HashSet[A]
   with Patchable[PatchableSet[A], SetPatch[A]] {
   override def applyPatch(patch: SetPatch[A]): PatchableSet[A] = {
-    val returnedSet = PatchableSet(this.toList :_*)
     this.add(patch.patchValue)
-    returnedSet.add(patch.patchValue)
-    returnedSet
+    PatchableSet(this.toList: _*)
```

Comment:
Yes, you are. You've now returned a set that may contain more changes than just what the patch did, which in some cases can cause incorrect behavior.